src/ui_basic/window.cc

```diff
@@ -384,8 +384,8 @@ void Window::move_inside_parent() {
 		} else {
 			if (py < toolbar_top_h) {
 				py = toolbar_top_h;
-			} else if (py + get_h() > ph) {
-				py = ph - get_h();
+			} else if (py + get_h() > ph + toolbar_top_h + toolbar_bottom_h) {
+				py = ph - get_h() + toolbar_top_h + toolbar_bottom_h;
```

Comment:
Only top. This is actually a coordinate transformation for `py`: The panel height is subtracted from `ph` to make it the effective usable height for comparisons with the window height. But the panel is also inside the parent, so the "effective" y coordinate (distance from the bottom of the panel in case of panel on top) is `toolbar_top_h` less than `py`. (Lines 385-386 handle the same issue for the "window above the effective top" case.)

```suggestion
			} else if (py + get_h() > ph + toolbar_top_h) {
				py = ph - get_h() + toolbar_top_h;
```

Reply:
Or maybe this one's better:

```suggestion
			} else if (const int32_t p_bottom = parent->get_inner_h() - toolbar_bottom_h; py + get_h() > p_bottom) {
				py = p_bottom - get_h();
```